src/dev/build/tasks/fleet_download_elastic_gpg_key.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Task, downloadToDisk } from '../lib';
+
+const BUNDLED_KEYS_DIR = 'x-pack/plugins/fleet/target/keys';
+const ARTIFACTS_URL = 'https://artifacts.elastic.co/';
+const GPG_KEY_NAME = 'GPG-KEY-elasticsearch';
+const GPG_KEY_SHA512 =
+  '84ee193cc337344d9a7da9021daf3f5ede83f5f1ab049d169f3634921529dcd096abf7a91eec7f26f3a6913e5e38f88f69a5e2ce79ad155d46edc75705a648c6';
+
+export const FleetDownloadElasticGpgKey: Task = {
+  description: 'Downloading Elastic GPG key for Fleet',
+
+  async run(config, log, build) {
+    const gpgKeyUrl = ARTIFACTS_URL + GPG_KEY_NAME;
+    const destination = build.resolvePath(BUNDLED_KEYS_DIR, GPG_KEY_NAME);
+    log.info(`Downloading Elastic GPG key from ${gpgKeyUrl} to ${destination}`);
+
+    try {
+      await downloadToDisk({
+        log,
+        url: gpgKeyUrl,
+        destination,
+        shaChecksum: GPG_KEY_SHA512,
+        shaAlgorithm: 'sha512',
+        skipChecksumCheck: false,
+        maxAttempts: 3,
+      });
+    } catch (error) {
+      log.warning(`Failed to download Elastic GPG key`);
```

Comment:
@kpollich I suppress the error downloading the key here, which is the same as the bundled packages. But I'm in two minds about it, should the build fail if we can't get the key?

Reply:
I don't think the build should fail, but we should disable package verification is this key is missing from disk. I think because we have to download this over the network, tying build success to this task is probably unwise just  because of network flakiness, etc.

Reply:
I would +1 failing the build if there's an expectation that this is part of a release.  I'd rather a flaky build than accidentally releasing without an expected file.

Reply:
Yeah, I'm tempted to side with @jbudz here. If the download fails in CI then disabling package verification would just fail a bunch of functional tests, right? Ideally we would just retry the download a couple times, and if it's a more permanent failure then maybe we would be able to retry the whole build... Without that error we wouldn't be able to recover and the whole CI run would fail. Additionally, if there aren't functional tests which rely on package verification, then there's definitely a chance we would release a build without the key in it.

Reply:
Fair enough, @jbudz and @spalger. Retrying the download piece and allowing CI to fail if we can't resolve this key sound like the right choices here. Thanks for contextualizing. 

> If the download fails in CI then disabling package verification would just fail a bunch of functional tests, right?

Ideally, yes. @hop-dev let's make sure we have some kind of API functional test with a test package + signature generated with the Elastic public key. If the key is missing or change in the CI environment, these tests should fail as the test package won't be considered valid. 

These tests don't need to be part of this PR, obviously. Once we have the package verification logic wired up let's tackle them.

Reply:
Thanks for the input, I've made it so the build will fail if we fail to download the key in [1e2f48d](https://github.com/elastic/kibana/pull/134861/commits/1e2f48d06ef02c5f48bcf7e38c79fffd4881352e).

Kyle I agree about the test, I will add an item to the implementation list for this 👍